forge_tests/TestLaunchProject.sol

```diff
@@ -0,0 +1,138 @@
+            dataSource: address(0),
+            metadata: 0
+        });
+    }
+
+    function testLaunchProject() public {
+        // Package up cycle config.
+        JBFundingCycleConfig[] memory _cycleConfig = new JBFundingCycleConfig[](1);
+        _cycleConfig[0].mustStartAtOrAfter = 0;
+        _cycleConfig[0].data = _data;
+        _cycleConfig[0].metadata = _metadata;
+        _cycleConfig[0].groupedSplits = new JBGroupedSplits[](0);
+        _cycleConfig[0].fundAccessConstraints = new JBFundAccessConstraints[](0);
+
+        // Package up terminal config.
+        JBTerminalConfig[] memory _terminalConfigurations = new JBTerminalConfig[](1);
+        JBAccountingContextConfig[] memory _accountingContexts = new JBAccountingContextConfig[](1);
+        _accountingContexts[0] =
+            JBAccountingContextConfig({token: JBTokens.ETH, standard: JBTokenStandards.NATIVE});
+        _terminalConfigurations[0] =
+            JBTerminalConfig({terminal: _terminal, accountingContextConfigs: _accountingContexts});
+
+        uint256 projectId = _controller.launchProjectFor({
+            owner: _projectOwner,
+            projectMetadata: _projectMetadata,
+            fundingCycleConfigurations: _cycleConfig,
+            terminalConfigurations: _terminalConfigurations,
+            memo: ""
+        });
+
+        // Get a reference to the first funding cycle.
+        JBFundingCycle memory fundingCycle = jbFundingCycleStore().currentOf(projectId);
+
+        // Make sure the funding cycle got saved correctly.
+        assertEq(fundingCycle.number, 1);
+        assertEq(fundingCycle.weight, _data.weight);
+    }
+
+    function testLaunchProjectFuzzWeight(uint256 _weight) public {
+        uint256 _projectId;
+        
+        _data = JBFundingCycleData({
+            duration: 14,
+            weight: _weight,
+            discountRate: 450_000_000,
+            ballot: IJBFundingCycleBallot(address(0))
+        });
+
+        // Package up cycle config.
+        JBFundingCycleConfig[] memory _cycleConfig = new JBFundingCycleConfig[](1);
+        _cycleConfig[0].mustStartAtOrAfter = 0;
+        _cycleConfig[0].data = _data;
+        _cycleConfig[0].metadata = _metadata;
+        _cycleConfig[0].groupedSplits = new JBGroupedSplits[](0);
+        _cycleConfig[0].fundAccessConstraints = new JBFundAccessConstraints[](0);
+
+        // Package up terminal config.
+        JBTerminalConfig[] memory _terminalConfigurations = new JBTerminalConfig[](1);
+        JBAccountingContextConfig[] memory _accountingContexts = new JBAccountingContextConfig[](1);
+        _accountingContexts[0] =
+            JBAccountingContextConfig({token: JBTokens.ETH, standard: JBTokenStandards.NATIVE});
+        _terminalConfigurations[0] =
+            JBTerminalConfig({terminal: _terminal, accountingContextConfigs: _accountingContexts});
+
+        // expectRevert on the next call if weight overflowing
+        if (_weight > type(uint88).max) {
```

Comment:
Fuzzing-wise, this creates a huge skew -> wdyt of rather use bound here (_weight ∈ [0; uint88.max -1] ) and have a small other test for > 2^88)?
-> rn, a lots of the fuzzed weight are falling in [2^88; 2^256], as forge fuzzer isn't random (or pure gaussian), it's (iirc) just a big kurtosis, here, we "loose" all the runs from both the mean and the right fat tail (most of the info for us is in the < 2^88 interval)

Reply:
If you see this pattern somewhere else, this should be part of overall refactoring (same as using vm.assume to select interval, for the same reason as vm.assume just drop the run while bound recenters it)

Reply:
Looks good but I think should be (_weight ∈ [0; uint88.max])